src/components/InLineNotification.tsx

```diff
@@ -0,0 +1,158 @@
+}
+
+interface CustomStyles {
+  container: ViewStyle
+  attentionIcon: Record<'color', Colors>
+  ctaLabel: TextStyle
+  ctaLabel2: TextStyle
+}
+
+export function InLineNotification({
+  severity,
+  title,
+  description,
+  style,
+  ctaLabel,
+  onPressCta,
+  ctaLabel2,
+  onPressCta2,
+  testID,
+}: Props) {
+  const severityStyle = severityStyles[severity]
+  return (
+    <View style={[defaultStyles.container, severityStyle.container, style]} testID={testID}>
+      {title && (
+        <View style={defaultStyles.row}>
+          <View style={defaultStyles.attentionIcon}>
+            <AttentionIcon color={severityStyle.attentionIcon.color} />
+          </View>
+          <Text style={defaultStyles.titleText}>{title}</Text>
+        </View>
+      )}
+
+      <View style={defaultStyles.bodyContainer}>
+        <View style={[defaultStyles.row, defaultStyles.bodyRow]}>
+          <View style={defaultStyles.attentionIcon}>
+            {!title && <AttentionIcon color={severityStyle.attentionIcon.color} />}
+          </View>
+          <Text style={defaultStyles.bodyText}>{description}</Text>
+        </View>
+
+        {(ctaLabel || ctaLabel2) && (
+          <View style={[defaultStyles.row, defaultStyles.ctaRow]}>
+            {ctaLabel && onPressCta && (
+              <Text style={[defaultStyles.ctaLabel, severityStyle.ctaLabel]} onPress={onPressCta}>
+                {ctaLabel}
+              </Text>
+            )}
+            {ctaLabel2 && onPressCta2 && (
+              <Text
+                style={[defaultStyles.ctaLabel2, severityStyle.ctaLabel2]}
+                onPress={onPressCta2}
+              >
+                {ctaLabel2}
+              </Text>
+            )}
+          </View>
+        )}
+      </View>
+    </View>
+  )
+}
+
+const defaultStyles = StyleSheet.create({
+  container: {
+    paddingHorizontal: Spacing.Regular16,
+    paddingTop: Spacing.Regular16,
+    paddingBottom: Spacing.Small12,
+    borderRadius: Spacing.Regular16,
+    width: '100%',
+    gap: Spacing.Tiny4,
+  },
+  bodyContainer: {
+    gap: Spacing.Smallest8,
+  },
+  row: {
+    flexDirection: 'row',
+    gap: Spacing.Smallest8,
+    alignItems: 'center',
+  },
+  bodyRow: {
+    alignItems: 'flex-start',
+  },
+  ctaRow: {
+    paddingVertical: Spacing.Tiny4,
+    paddingHorizontal: Spacing.Smallest8,
+    justifyContent: 'flex-end',
+  },
+  attentionIcon: {
+    padding: Spacing.Tiny4,
+    width: Spacing.Thick24,
+    height: Spacing.Thick24,
+  },
+  titleText: {
+    ...fontStyles.small600,
+  },
+  bodyText: {
+    ...typeScale.bodyXSmall,
+    lineHeight: 18,
+    color: Colors.dark,
+    flexShrink: 1,
+  },
+  ctaLabel: {
+    ...fontStyles.small600,
+    paddingHorizontal: Spacing.Smallest8,
+    paddingVertical: Spacing.Tiny4,
+  },
+  ctaLabel2: {
+    ...fontStyles.small600,
+    paddingHorizontal: Spacing.Smallest8,
+    paddingVertical: Spacing.Tiny4,
+  },
+})
+
+const severityStyles: Record<Severity, CustomStyles> = {
```

Comment:
this works, but i think we may not need so much complexity / flexibility. providing a way to change many styles for these elements implies that we can customise the component a lot between the different severity levels (e.g. i can maybe add a paddingVertical to the container in Information, but not in Warning) - i think that this is flexibility that we actually don't want, to keep the component looking consistent between the severity levels. any customised style props should be applied to all variations.

from the design, it looks like the only thing that changes between the severity levels is color. so i think we could define a simple map that provides a primary and secondary color to each severity, and just add inline styles inside the component (e.g. instead of ` <Text
                style={[defaultStyles.ctaLabel2, severityStyle.ctaLabel2]}
                onPress={onPressCta2}
              >` we could have ` <Text
                style={[defaultStyles.ctaLabel2, { color: inLineNotificationColors[severity].primary }]}
                onPress={onPressCta2}
              >` . i'd argue that this makes the component easier to read, since the level of customisation is right there. what do you think?

Reply:
>can customise the component a lot between the different severity levels 
Originally I meant it for customising colours, e.g. if CTA buttons may need to have different colors. If it's needed in the future, we can come back. I agree on keeping things consistent with Severity (or Type/Level).

Reply:
I'm missing the word `color` either in the name of the object (e.g. `severetyColors`) or in property names (e.g. `primaryColor`, `secondaryColor`).

Reply:
Changed to the namings `CustomColors`, `severityColor`, `severityColors` in https://github.com/valora-inc/wallet/pull/4436/commits/3c25faf35a9a558099c54822172b446e10aaaaf3